libs/langchain/langchain/utils/iter.py

```diff
@@ -161,3 +163,13 @@ def close(self) -> None:
 
 # Why this is needed https://stackoverflow.com/a/44638570
 safetee = Tee
+
+
+def batch(size: int, iterable: Iterable[T]) -> Iterator[List[T]]:
```

Comment:
is batch too generic? maybe `batch_iterate`?

Reply:
it's under the `iter` namespace?

Reply:
I added as `batch_iterate` -- we can always rename to `batch` afterwards or keep as is, doesn't matter that much to me